File: university-service/src/main/java/ua/com/foxminded/courseproject/validation/AgeValidator.java
package ua.com.foxminded.courseproject.validation;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

public class AgeValidator implements ConstraintValidator<Age, LocalDate> {
    private long min;

    @Override
    public void initialize(Age constraintAnnotation) {
        ConstraintValidator.super.initialize(constraintAnnotation);
        this.min = constraintAnnotation.min();
    }

    @Override
    public boolean isValid(LocalDate birthday, ConstraintValidatorContext context) {
        if (birthday == null) {
            return true;
        }
        long age = LocalDate.from(birthday).until(LocalDate.now(), ChronoUnit.YEARS);
        return age >= min;
    }

}

File: university-service/src/main/java/ua/com/foxminded/courseproject/validation/AgeValidator.kt
package ua.com.foxminded.courseproject.validation

import java.time.LocalDate
import java.time.temporal.ChronoUnit
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class AgeValidator : ConstraintValidator<Age, LocalDate?> {
    private var min: Long = 0
    override fun initialize(constraintAnnotation: Age) {
        super.initialize(constraintAnnotation)
        this.min = constraintAnnotation.min
    }

    override fun isValid(birthday: LocalDate?, context: ConstraintValidatorContext): Boolean {
        if (birthday == null) {
            return true
        }
        val age = LocalDate.from(birthday).until(LocalDate.now(), ChronoUnit.YEARS)
        return age >= min
    }
}
